Setup dockerfile and artifact text for artifact evaluation (#162)
opencompl/ssa: artifact-evaluation/artifact.tex

%   Michael Heroux, Michela Taufer and other colleagues.
%
% See examples of this Artifact Appendix in
%  * SC'17 paper: https://dl.acm.org/citation.cfm?id=3126948
%  * CGO'17 paper: https://www.cl.cam.ac.uk/~sa614/papers/Software-Prefetching-CGO2017.pdf
%  * ACM ReQuEST-ASPLOS'18 paper: https://dl.acm.org/citation.cfm?doid=3229762.3229763
%
% (C)opyright 2014-2022
%
% CC BY 4.0 license
%

\documentclass{sigplanconf}

\usepackage{hyperref}
\usepackage{minted}

\begin{document}

\special{papersize=8.5in,11in}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% When adding this appendix to your paper, 
% please remove above part
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\appendix
\section{Artifact Appendix}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Abstract}

{\em Obligatory}

\subsection{Artifact check-list (meta-information)}

% {\em Obligatory. Use just a few informal keywords in all fields applicable to your artifacts
% and remove the rest. This information is needed to find appropriate reviewers and gradually 
% unify artifact meta information in Digital Libraries.}

{\small
\begin{itemize}

>   \item {\bf Program: } The code repsitory for our framework along with the test suite.

Reviewer: ```suggestion
  \item {\bf Program: } The code repository for our framework along with the test suite.
```

  \item {\bf Compilation: } Python for running the plotting scripts and the Lean4 toolchain, downloaded via \texttt{elan}
  \item {\bf Run-time environment: } Any operating system that supports Docker.
  \item {\bf Hardware: } Any x86-64 machine.
  \item {\bf Output: } Key theorems of the paper will be built and shown to have no unsound axioms.
  \item {\bf How much disk space required (approximately)?: } 10GB
  \item {\bf How much time is needed to prepare workflow (approximately)?: } 1hr
  \item {\bf How much time is needed to complete experiments (approximately)?: } 1hr
  \item {\bf Publicly available?: } Yes
  \item {\bf Code licenses (if publicly available)?: } MIT
  \item {\bf Archived (provide DOI)?: }
\end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Description}

\subsubsection{Hardware dependencies}

None.

\subsubsection{Software dependencies}

> Docker is necessary to run our artifact. the Docker image has dependences needed to compile our framework with Lean4.

Reviewer: ```suggestion
Docker is necessary to run our artifact. The Docker image has all dependencies needed to compile our framework with Lean4.
```

%COMMENT% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%COMMENT% \subsection{Installation}
%COMMENT% 
%COMMENT% {\em Obligatory}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Experiment workflow}

Access the docker image \texttt{opencompl-ssa} from
(\url{http://doi.org/TODO-TODO-TODO-DOI}), then run:

\begin{minted}[fontsize=\scriptsize]{text}
$ docker load -i opencompl-ssa.docker
$ docker run -it siddudruid/opencompl-ssa
# | This clears the build cache,
# | fetches the maths library from the build cache,

> # v and builds our framework.

Reviewer: ```suggestion
# | and builds our framework.
```
